lib/group_admin.js

```diff
@@ -38,6 +39,82 @@ GroupAdmin.prototype.describeGroup = function (groupId) {
 };
 
 /**
+ * Provides output similar to
+ * kafka.admin.ConsumerGroupCommand --describe
+ *
+ * @param  {String} groupId
+ * @param  {Array} offsetFetchRequestArgs
+ * @return {Promise}
+ */
+GroupAdmin.prototype.fetchConsumerLag = function (groupId, offsetFetchRequestArgs) {
+    return Promise.all([
+        this.client.offsetFetchRequestV1(groupId, offsetFetchRequestArgs),
+        this._fetchHighWaterMark(offsetFetchRequestArgs)
+    ]).spread(function (offsets, highWaterMark) {
+        var i, j, res;
+        var responses = [];
+
+        // Match the returned data into a coherent response
+        for (i = 0; i < offsets.length; i++) {
+            res = {
+                topic: offsets[i].topic,
+                partition: offsets[i].partition,
+                offset: offsets[i].offset,
+                highwaterMark: null,
+                consumerLag: null
+            };
+
+            for (j = 0; j < highWaterMark.length; j++) {
+                if (
+                    offsets[i].topic === highWaterMark[j].topic
+                    && offsets[i].partition === highWaterMark[j].partition
+                ) {
+                    res.highwaterMark = highWaterMark[j].highWaterMark;
+                    if (res.offset > 0 && res.highwaterMark > 0) {
+                        res.consumerLag = res.highwaterMark - res.offset;
+                    }
+                }
+            }
+
+            responses.push(res);
+        }
+
+        return responses;
+    });
+};
+
+GroupAdmin.prototype._fetchHighWaterMark = function (offsetFetchRequestArgs) {
+    var topicName, topicPartitions, i, j;
+    var offsetRequestsPromises = [];
+
+    for (i = 0; i < offsetFetchRequestArgs.length; i++) {
+        topicName = offsetFetchRequestArgs[i].topicName;
+        topicPartitions = offsetFetchRequestArgs[i].partitions;
+
+        for (j = 0; j < topicPartitions.length; j++) {
+            offsetRequestsPromises.push(
+                this._fetchHighWaterMarkOffset(topicName, topicPartitions[j])
+            );
+        }
+    }
+
+    return Promise.all(offsetRequestsPromises);
```

Comment:
The `Client` won't do any queueing, a per-leader `Connection` instance will do some sort of queue, but it definitely will grow memory buffers rapidly. I don't think Promise.map with concurrency limit of 10 will have any difference for topics with small number of partitions, but it may save from extra memory operations on large topics.